src/tic-tac-toe.ts

```diff
@@ -40,36 +44,98 @@ class Game {
     return square === this.currentPlayer;
   };
 
-  private checkRowsForWin(): boolean {
-    return this.board.some(row => row.every(this.isCurrentPlayer));
+  private checkRowsForWin(): WinData {
+    const winData: WinData = {
+      hasWinner: false,
+      winningCells: [],
+    };
+
+    for (const rowIndex in this.board) {
+      if (this.board[rowIndex].every(cell => this.isCurrentPlayer(cell))) {
+        winData.hasWinner = true;
+        for (
+          let columnIndex = 0;
+          columnIndex < this.board.length;
+          columnIndex++
+        ) {
+          winData.winningCells.push([Number(rowIndex), columnIndex]);
+        }
+        break;
+      }
+    }
+
+    return winData;
   }
 
-  private checkColumnsForWin(): boolean {
+  private checkColumnsForWin(): WinData {
+    const winData: WinData = {
+      hasWinner: false,
+      winningCells: [],
+    };
+
     const transposedBoard = this.eachIndex().map(i =>
       this.board.map(row => row[i])
     );
 
-    return transposedBoard.some(row => row.every(this.isCurrentPlayer));
+    winData.hasWinner = transposedBoard.some(row =>
+      row.every(this.isCurrentPlayer)
+    );
+
+    if (winData.hasWinner) {
+      transposedBoard.forEach((row, rowIndex) => {
+        if (row.every(value => this.isCurrentPlayer(value))) {
+          const vectors = [];
+          for (let i = 0; i < row.length; i++) {
+            vectors.push([rowIndex, i]);
```

Comment:
If you changed this to `vectors.push([i, rowIndex])`, would that obviate the need for `transposeVectors` ?

Reply:
also,

```js
winData.winningCells = row.map((_, i) => [i, rowIndex])
```